zerver/tests/test_events.py

```diff
@@ -2692,6 +2712,19 @@ def do_upload() -> None:
         self.subscribe(hamlet, "Denmark")
         assert url is not None
         body = f"First message ...[zulip.txt](http://{hamlet.realm.host}" + url + ")"
+
+        # We explicitly set the 'automatically_follow_topics_policy' to
+        # 'AUTOMATICALLY_FOLLOW_TOPICS_NEVER', to make the test independent
+        # of 'automatically_follow_topics_policy'.
+        #
+        # The reason being that our main focus here is not to test the extra
+        # events generated depending on the 'automatically_follow_topics_policy'.
+        do_change_user_setting(
+            hamlet,
+            "automatically_follow_topics_policy",
+            UserProfile.AUTOMATICALLY_FOLLOW_TOPICS_NEVER,
+            acting_user=None,
+        )
```

Comment:
> I think we had decided to set the policy for test users to be NEVER in the test database? 

Yes, I finally did that.

Earlier, I set the policy in the individual tests themselves because, as [Steve said](https://chat.zulip.org/#narrow/stream/101-design/topic/automatically.20follow.20topics.20.2325914/near/1602472):
> Generally I have found it easier when writing (and reading) tests to just explicitly set up the data in individual tests. The set_up() methods can be kind of tricky for future developers who don't realize right away that there is action-at-a-distance.

But I agree that it is too verbose. So [Tim's suggestion](https://chat.zulip.org/#narrow/stream/101-design/topic/automatically.20follow.20topics.20.2325914/near/1609293) would be the better approach here:
>  we can set a defaults during the creation of test users in populate_db, though.